units/systemd-journald.service.in

```diff
@@ -15,6 +15,11 @@ Requires=systemd-journald.socket
 After=systemd-journald.socket systemd-journald-dev-log.socket systemd-journald-audit.socket syslog.socket
 Before=sysinit.target
 
+# To avoid journald SIGKILLed during soft-reboot and corrupting journals.
+# See https://github.com/systemd/systemd/issues/30195
+Before=soft-reboot.target
```

Comment:
Shouldn't this be `shutdown.target` instead, to cover regular reboot/shutdown too?

Reply:
Also, `systemd-journald@.service.in` likely needs this too.

Reply:
No, that's intended. On shutdown or regular reboot, systemd-journald will be killed by systemd-shutdown on SIGTERM.
systemd-shutdown wait for processes being terminated, so journald can safely sync and close journal files.

When shutdown.target is used here, then we cannot log anything about shutting down process.

Reply:
> Also, `systemd-journald@.service.in` likely needs this too.

Indeed. Thanks.